src/main/java/uk/gov/register/views/representations/spreadsheet/EntriesSpreadSheetWriter.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.register.views.representations.spreadsheet;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.ss.usermodel.Workbook;
+import uk.gov.register.util.HashValue;
+import uk.gov.register.views.EntryListView;
+import uk.gov.register.views.representations.ExtraMediaType;
+import uk.gov.register.views.representations.RepresentationWriter;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Provider
+@Produces(ExtraMediaType.TEXT_SPREADSHEET)
+public class EntriesSpreadSheetWriter extends RepresentationWriter<EntryListView> {
+
+    @Override
+    public void writeTo(final EntryListView entryListView, final Class<?> type, final Type genericType,
+                        final Annotation[] annotations, final MediaType mediaType,
+                        final MultivaluedMap<String, Object> httpHeaders, final OutputStream entityStream)
+            throws IOException, WebApplicationException {
+        final List<String> fieldNames = new ArrayList<>();
+        final List<Map<String, String>> elements = new ArrayList<>();
+        final Workbook workbook;
+
+        fieldNames.add("index-entry-number");
+        fieldNames.add("entry-number");
+        fieldNames.add("entry-timestamp");
+        fieldNames.add("key");
+        fieldNames.add("item-hash");
+
+        entryListView.getEntries().forEach(entry -> {
+            final Map<String, String> element = new HashMap<>();
+            String itemHashValues = StringUtils.EMPTY;
+
+            element.put("index-entry-number", entry.getIndexEntryNumber().toString());
+            element.put("entry-number", entry.getEntryNumber().toString());
+            element.put("entry-timestamp", entry.getTimestampAsISOFormat());
+            element.put("key", entry.getKey());
+
+            if (!entry.getItemHashes().isEmpty()) {
+                itemHashValues = entry.getItemHashes()
```

Comment:
@fjavierm I may have told you the wrong thing here. @michaelabenyohai what do we want to do in a spreadsheet if there are either multiple or no item hashes for an entry? The specification does not specify for this format, nor for csv or tsv.

Reply:
After checking the existing code and the output of CSV files with multiple items per entry, I believe this code is correct. See the attached screenshots.

*Existing code rendering `csv`*

<img width="1191" alt="screen shot 2017-09-28 at 14 28 22" src="https://user-images.githubusercontent.com/5422487/30969245-b81eb88e-a459-11e7-9262-9e82ad8e69e9.png">

*New code rendering `xlsx`*

<img width="1250" alt="screen shot 2017-09-28 at 14 28 04" src="https://user-images.githubusercontent.com/5422487/30969297-d9964978-a459-11e7-93dc-96741d0bd095.png">